src/package-request.js

```diff
@@ -136,6 +138,30 @@ export default class PackageRequest {
   }
 
   /**
+   * If pattern:
+   * - contains : it will return because is something like '<protocol>:<something else' or an url
+   * - is a file it will prepend file:
+   * otherwise it will return
+   */
+  async normalizeRange(pattern: string): Promise<string> {
+    if (pattern.includes(':')) {
+      return Promise.resolve(pattern);
+    }
+
+    if (await fs.exists(path.join(this.config.cwd, pattern))) {
```

Comment:
How often is this IO operation is used?
Can we infer if pattern looks like a file first before touching the disk if this happens often?